src/glu/problem_description.cpp

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/datatype.hpp>
+#include <miopen/glu/problem_description.hpp>
+#include <miopen/names.hpp>
+
+#include <sstream>
+
+namespace miopen {
+
+namespace glu {
+
+NetworkConfig ProblemDescription::MakeNetworkConfig() const
+{
+    switch(direction)
+    {
+    case Direction::Forward: return MakeForwardNetworkConfig();
+    case Direction::Backward: return MakeBackwardNetworkConfig();
+    default: MIOPEN_THROW(miopenStatusInternalError);
+    }
+}
+
+NetworkConfig ProblemDescription::MakeForwardNetworkConfig() const
```

Comment:
Let's remove these functions from the class and put them into unnamed namespace

Reply:
I realized that strings generated by `MakeForwardNetworkConfig()` and `MakeBackwardNetworkConfig()` share the same structure, so generating this structure directly in `MakeNetworkConfig()` is sufficient.